src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -41,6 +42,9 @@ public static class BitOperations
             19, 27, 23, 06, 26, 05, 04, 31
         };
 
+        private static readonly uint[] s_crcTable = Sse42.IsSupported || Crc32.IsSupported ?
```

Comment:
```suggestion
        private static readonly uint[] s_crcTable = Sse42,X64.IsSupported || Crc32.Arm64.IsSupported ?
```

Reply:
Thank you

Reply:
This is not a good fix, because it means the LUT will always be created on x86, even if it is never used.  The `ulong` overload will also always be processed byte-by-byte with the fallback implementation, when it would be *much* faster to decompose to 2 `uints` and call the intrinsic.

Reply:
Do you have an idea to provide a better fix? Can't something like this be optimized by the R2R "compiler"?

Reply:
The potential R2R complication I was referring to in a previous comment was related to this: https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/vectors-and-intrinsics.md#code-review-rules-for-code-written-in-systemprivatecorelibdll

The rules are different under `crossgen2`, so I don't believe that's a problem anymore.

You've got 2 problems left:
1) x86 clients will get the slow software fallback if they call the `ulong` overload, when it would be faster to break it down and call the `uint` intrinsic twice.  You can fix this by adding a 32-bit `Sse42.Crc32` fallback path when `Sse42.X64` is not supported but `Sse42` is.
2) If neither the ARM nor SSE4.2 intrinsic is supported, the CRC LUT will be created as part of the `BitOperations` class initializer.  Since other `BitOperations` methods are called from various places in the BCL, this means the LUT will always be created regardless of whether `BitOperations.Crc32C` is ever used.  The two ways to fix that would be:
a) Lazy-initialize the LUT only when it's needed, or
b) Move the static LUT field to its own static class, so that the initializer for that class only runs if the fallback LUT is actually needed.
The lazy-initialization option would incur a cost on each call to `Crc32C`, so it would likely be more desirable to take the latter option.

Reply:
> If neither the ARM nor SSE4.2 intrinsic is supported,

this scenario (someone on ancient computer getting slightly degraded performance) is not something we should worry about when rest of the runtime doesn't either.

Reply:
It's not just about ancient hardware.  WASM doesn't support either ISA, so it would be included.

Reply:
one kb lut field, which we are talking about here, is drop in a bucket compared to what a simplest wasm app's startup loads. 
but why not, if you think calling intrinsic on decontructed input values and reconstructing the output for supported endianess will produce the same result more efficiently, feel free to suggest the code change like i did above.

Reply:
To use your metaphor, with enough drops, the bucket will overflow.  I won't speak for the BCL team, but it's clear they have made a concerted effort to reduce startup overhead in the last couple of .NET releases.  If they want to take this PR as is, that's fine -- I'm just pointing out an easily avoidable overhead this would introduce.

> feel free to suggest the code change like i did above

It wouldn't be a one-line change, and this is not my PR.  Leaving it as a learning exercise for @deeprobin

Reply:
Yes I still have a lot to learn :D. In this regard, a quote from Winston Churchill.

> I am always ready to learn although I do not always like being taught.